web3swift/src/Extensions/KeccakExtensions.swift

```diff
@@ -33,3 +33,20 @@ public extension Web3Extensions where Base == String {
         return data.web3.keccak256
     }
 }
+
+public extension String {
+    func toChecksumAddress() -> String {
+        let lowerCaseAddress = self.stripHexPrefix().lowercased()
+        let arr = Array(lowerCaseAddress)
+        let keccaf = Array(lowerCaseAddress.web3.keccak256.web3.hexString.stripHexPrefix())
+        var result = "0x"
+        for i in 0 ... lowerCaseAddress.count - 1 {
+            if let val = Int(String(keccaf[i]), radix: 16), val >= 8 {
+                result.append(arr[i].uppercased())
+            } else {
+                result.append(arr[i])
+            }
+        }
+        return result
+    }
```

Comment:
This should be an method in `EthereumAddress`

Reply:
Done